libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline/validations/relations.rs

```diff
@@ -190,11 +190,16 @@ pub(super) fn referencing_fields_in_correct_order(
 /// relation links to NoAction for both referential actions.
 pub(super) fn cycles<'ast, 'db>(
     relation: CompleteInlineRelationWalker<'ast, 'db>,
-    connector: &dyn Connector,
+    db: &'db ParserDatabase<'ast>,
     diagnostics: &mut Diagnostics,
 ) {
-    if !connector.has_capability(ConnectorCapability::ReferenceCycleDetection)
-        || !connector.has_capability(ConnectorCapability::ForeignKeys)
+    if !db
+        .active_connector()
+        .has_capability(ConnectorCapability::ReferenceCycleDetection)
+        || db
+            .datasource()
+            .map(|ds| ds.referential_integrity().is_prisma())
```

Comment:
This check we're going to change on this week. Let's still keep it like this though.